currency-conversion/src/main/java/lk/ac/ucsc/currencyconversion/CurrencyConversionController.java

```diff
@@ -0,0 +1,47 @@
+package lk.ac.ucsc.currencyconversion;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+public class CurrencyConversionController {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    private ForexServiceProxy forexService;
+
+
+    @GetMapping("/currency-converter/from/{from}/to/{to}/quantity/{quantity}")
+    public CurrencyConversionBean convertCurrency(@PathVariable String from, @PathVariable String to,
+                                                  @PathVariable double quantity) {
+
+//        Map<String, String> uriVariables = new HashMap<>();
+//        uriVariables.put("from", from);
+//        uriVariables.put("to", to);
+//
+//        ResponseEntity<CurrencyConversionBean> responseEntity = new RestTemplate().getForEntity(
+//                "http://localhost:8000/currency-exchange/from/{from}/to/{to}", CurrencyConversionBean.class,
+//                uriVariables);
+//
+//        CurrencyConversionBean response = responseEntity.getBody();
+
+        CurrencyConversionBean response = forexService.retrieveExchangeValue(from, to);
+
+        return new CurrencyConversionBean(response.getId(), from, to, response.getConversionRate(), quantity,
```

Comment:
@Binurim u can directly pass 'response' into constructor and assign id,convRate and port from inside it.
unless u need the native all args contructor in other places.

Reply:
Need help here